Updated functionalities and code for "projects"

AleksandarAngelov01 opened the pull request:
Fixed all of the issues after the first review. Still no validation added.

AleksandarAngelov01 committed:
ProjectsController - Create Project, Delete Project, Edit project

AleksandarAngelov01 committed:
Changes after review

Fixed Service, Controller and Model +++

beval1 commented on src/main/java/com/taskmaster/server/projects/ProjectsController.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @PostMapping("/projects")
+    public ResponseEntity<?> createProject(@RequestBody ProjectsDto dto) {
```
you can return ResponseDTO, it's standartized response with message and status

beval1 commented on src/main/java/com/taskmaster/server/projects/ProjectsController.java:
```diff
@@ -0,0 +1,55 @@
+    @GetMapping("/{projectId}")
+    public ResponseEntity<?> getProjectById(@PathVariable Long projectId) {
+        Optional<ProjectsModel> project = projectsService.findProjectById(projectId);
+        return project.map(value -> new ResponseEntity<>(value, HttpStatus.OK))
```
this logic should be in the service layer

beval1 commented on src/main/java/com/taskmaster/server/projects/ProjectsMapper.java:
```diff
@@ -0,0 +1,22 @@
+        this.modelMapper = modelMapper;
+    }
+
+    public ProjectsModel convertDtoToEntity(ProjectsDto dto, String name) {
```
why the name is not part of the dto but passed as argument?

beval1 commented on src/main/java/com/taskmaster/server/projects/ProjectsMapper.java:
```diff
@@ -0,0 +1,22 @@
+
+@Component
+public class ProjectsMapper {
+
```
Delete this file. You can move this method inside the service or even in the Dto itself

beval1 commented on src/main/java/com/taskmaster/server/projects/ProjectsDto.java:
```diff
@@ -0,0 +1,5 @@
+package com.taskmaster.server.projects;
+
+public record ProjectsDto(Long id, String name) {
```
Do we expect to receive a project with just a named? You are checking in the service class if the id is null, why not use "long" primitive here and remove the check?

AleksandarAngelov01 committed:
Changes in GlobalExceptionHandler

Added exceptions

viliana0952 committed:
Added User

Service, Controller, Model, Mapper, DTO, Repository

AleksandarAngelov01 committed:
New Entities added and exceptions

Almost ready with the set up for projects and users and the connection between them.

AleksandarAngelov01 committed:
Started relations

beval1 committed:
fixed up all the code
added project categories
added project memberships
security validation on everything
